promql/engine.go

```diff
@@ -1498,17 +1521,20 @@ func (ev *evaluator) matrixIterSlice(it *storage.BufferedSeriesIterator, mint, m
 			}
 			out = append(out, Point{T: t, V: v})
 			ev.currentSamples++
+			ev.samplesStats.UpdatePeak(ev.currentSamples)
```

Comment:
Can we do this outside the for? That'll save the conditional branch in the hot path.

This can be done in other places too.

Reply:
Yes, that will be better.

Reply:
Also, do note that in some parts, the UpdatePeak is present inside the loop so that if any errors occur, then the information about peak samples is there as well.

Reply:
Done.